Update interactions.tsv

debpaul opened the pull request:
please review @jhpoelen

- did: added OBO RO interactionTypeId where match found in ontology. 
- quotes and encoding: also note that I specified UTF-8, but don't see a way in Windows Excel to "disable quotes"
- No direct matches found for these terms: kleptoplastidy, epibiosis, allelopathy, competes with

debpaul committed:
Update interactions.tsv

added OBO RO interactionTypeId where match found in ontology. No direct matches found for these terms: kleptoplastidy
epibiosis
allelopathy
competes with

jhpoelen commented on interactions.tsv:
```diff
@@ -1355,50 +1355,50 @@ sourceOccurrenceId	sourceTaxonId	sourceTaxonName	sourceBodyPartId	sourceBodyPart
+		Epithemia turgida					http://purl.obolibrary.org/obo/RO_0002453	host of			EtSB 										10.1073/pnas.1405222111	(Nakayama et al. 2014)
+		Asterionella formosa					http://purl.obolibrary.org/obo/RO_0002445	has parasite			Chytrids										10.1038/ismej.2013.73	(Gsell et al. 2013)
+		Rhizosolenia hebetata					http://purl.obolibrary.org/obo/RO_0002453	host of			Richelia intracellularis										WOS:000325596800006	(Madhu et al. 2013)
+		Pseudo nitzschia 					http://purl.obolibrary.org/obo/GO_0085030	mutualism			Bacteria										10.1038/ismej.2013.138	(Sison-Mangus et al.2014)
```
Rather that using http://purl.obolibrary.org/obo/GO_0085030 , suggest to use http://purl.obolibrary.org/obo/RO_0002442 .

debpaul replied:
Got it. I'll make the changes in my local copy.
Do I make a new pull request afterwards?
Never done this specific workflow before :-)

jhpoelen commented on interactions.tsv:
```diff
@@ -23,1329 +23,1329 @@ sourceOccurrenceId	sourceTaxonId	sourceTaxonName	sourceBodyPartId	sourceBodyPart
-	EOL:3685	Bacillariophyta 						eaten by		EOL:213022	Opisthonema oglinum 											
-	EOL:3685	Bacillariophyta 						eaten by		EOL:54923	Polypedilum 											
-	EOL:3685	Bacillariophyta 						eaten by		EOL:55368	Naonella 											
-	EOL:3685	Bacillariophyta 						eaten by		EOL:481634	A
```
Rather than using http://purl.obolibrary.org/obo/RO_0002434  (interacts with) , suggest to use http://purl.obolibrary.org/obo/RO_0002437 (biotically interacts with)

debpaul replied:
makes sense to me.
making changes to my local copy now

jhpoelen reviewed (changes requested):
hey @debpaul - wow, this looks really nice, some minor comments on the usage of two terms. Please review my comments and apply changes. Let me know if you have any questions!

debpaul committed:
swapped out 2 interactionTypeId URIs

followed @jhpoelen suggesting to use
RO_0002442 (instead of GO_0085030) and RO_0002437 (instead of RO_0002434), fixed a missing parens too. Used notepad++ to try to avoid quotes issues

debpaul commented:
to @jhpoelen note that i used Notepad++ to edit this time, instead of Excel. Please check formatting and note if (at least some) of the extraneous quote marks are now gone.

jhpoelen commented on interactions.tsv:
```diff
@@ -897,42 +897,42 @@ sourceOccurrenceId	sourceTaxonId	sourceTaxonName	sourceBodyPartId	sourceBodyPart
-	EOL:902512	Cymbella aspera 					http://purl.obolibrary.org/obo/RO_0002434	interacts with		EOL:54832	Cricotopus 											
-	EOL:902512	Cymbella aspera 					http://purl.obolibrary.org/obo/RO_0002434	interacts with		EOL:52934	Pycnocentria 											
-	EOL:902512	Cymbella aspera 					http://purl.obolibrary.org/obo/RO_0002434	interacts with		EOL:9021	Tanyderidae 											
+	EOL:902512	Cymbella aspera 					http://purl.obolibrary.org/obo/RO_0002437	interacts with		EOL:19712339	no:match 											
```
@debpaul Nice!

jhpoelen commented on interactions.tsv:
```diff
@@ -1380,7 +1380,7 @@ sourceOccurrenceId	sourceTaxonId	sourceTaxonName	sourceBodyPartId	sourceBodyPart
 		Asterionella formosa					http://purl.obolibrary.org/obo/RO_0002445	has parasite			Chytrids										10.1038/ismej.2013.73	(Gsell et al. 2013)
 		Rhizosolenia hebetata					http://purl.obolibrary.org/obo/RO_0002453	host of			Richelia intracellularis										WOS:000325596800006	(Madhu et al. 2013)
-		Pseudo nitzschia 					http://purl.obolibrary.org/obo/GO_0085030	mutualism			Bacteria										10.1038/ismej.2013.138	(Sison-Mangus et al.2014)
+		Pseudo nitzschia 					http://purl.obolibrary.org/obo/RO_0002442	mutualism			Bacteria										10.1038/ismej.2013.138	(Sison-Mangus et al.2014)
```
@debpaul Thanks for using http://purl.obolibrary.org/obo/RO_0002442 instead. Your usage actually uncovered that GloBI didn't support the mutualist interaction type (e.g., from travis log: ```found unsupported interactionTypeId [http://purl.obolibrary.org/obo/RO_0002442]```). I've added support just now via https://github.com/jhpoelen/eol-globi-data/commit/da3ac13472795e93ee55dc74d56fc3e43deac73d . So, on the next GloBI release (within a couple of days), the mutualist interaction type will be available. In short: you helped improve not only this dataset, but GloBI itself!

debpaul commented:
So sad to see Travis fail :-) but so happy to know it's just because he's on-the-lookout for missing data and other issues. It's not my changes - thank goodness.

jhpoelen commented:
re: travis - I wish travis was a little more encouraging - like:

"Hey Deb, thanks for your improvement, and by the way, there's some other things that need attention if you have some time." 

Travis is traditionally used for running software integrity checks and has a green light / red light (pass/fail) idea to remove any ambiguity.

jhpoelen commented:
And . . . I reviewed your updated pull request and it looks great! 

Now, lets hope that @FloraVincent will merge your proposal (aka pull request) into her Diatom Interaction Database DIDB. Meanwhile, your forked repository will also be indexed by GloBI, so your contributions are visible.

Commit:
attempt to remove unnecessary quoting

I was able to remove the quotes by: 1. replace tabs with commas using "cat interactions.tsv | sed 's/\t/,/g' > interactions.csv" 2. re-import into LibreOffice calc as csv file 3. re-export to tsv using save as csv with tab delimiters, utf-8 character encoding and no string quoting.

debpaul committed:
Merge pull request #1 from jhpoelen/patch-2

attempt to remove unnecessary quoting
thanks @jhpoelen

Commit:
support to update citation - also see https://github.com/FloraVincent/DIDB/issues/1

debpaul committed:
Merge pull request #2 from jhpoelen/patch-3

support to update citation - also see https://github.com/FloraVincent…
